src/garage/tf/core/parameter.py

```diff
@@ -31,15 +34,14 @@ def parameter(input_var,
         A tensor of the broadcasted variables.
     """
     with tf.variable_scope(name):
-        p = tf.get_variable(
-            'parameter',
-            shape=(length, ),
-            dtype=dtype,
-            initializer=initializer,
-            trainable=trainable)
-
+        if param is None:
+            param = tf.get_variable(
+                'parameter',
```

Comment:
Are you sure this shouldn't be `name`? TBH, I'm not totally sure how this class is used.

Reply:
I should have given more context. Currently we use this parameter function for modeling a "parameter" - a reshaped tf.Variable so that it matches the input placeholder shape. Therefore we always call the tf.Variable `parameter`, similar to that the weights of a MLP is called `kernel`.

However, sometimes the underlying tf.Variable (not the reshaped tf.Variable) is needed to be reused, and should be returned as well. Similarly, the parameter function should have an argument for the incoming tf.Variable if it needs to be reused.

Reply:
i'm not sure what's going on here. are you using this as a helper function to broadcast an existing param? why are there now two return values?

Reply:
I am using to create a new param and also broadcast it. In some cases the same param is needed for another parameter so I return it as well.

Reply:
this is a little bit strange. all of the rest of the primitive functions imply resource allocation. suddenly this one may or may not allocate resources and also for some reason has 2 return values.

is it possible to refactor it into the original (which allocates) and a helper (which broadcasts)? are those the two relevant parts here?

if you supply me with the parameter, i don't need to return it back to you. this is python, you already have it. do you also need access to the allocated params?

Reply:
It actually makes more sense to separate into two different helper functions. Will do.